ide/o.n.swing.dirchooser/src/org/netbeans/swing/dirchooser/Module.java

```diff
@@ -96,17 +94,27 @@ private static void install() {
             uid.put(val, impl);
         }
         // #61147: prevent NB from switching to a different UI later (under GTK):
-        uid.addPropertyChangeListener(pcl = new PropertyChangeListener() {
-            public @Override void propertyChange(PropertyChangeEvent evt) {
-                String name = evt.getPropertyName();
-                Object className = uid.get(KEY);
-                if ((name.equals(KEY) || name.equals("UIDefaults")) && !val.equals(className)
-                        && !isQuickFileChooser(className)) {
-                    originalImpl = (Class<?>) uid.getUIClass(KEY);
-                    uid.put(KEY, val);
-                }
+        uid.addPropertyChangeListener(pcl = (evt) -> {
+            String name = evt.getPropertyName();
+            Object className = uid.get(KEY);
+            if ((name.equals(KEY) || name.equals("UIDefaults")) && !val.equals(className) && !isQuickFileChooser(className)) {
+                originalImpl = (Class<?>) uid.getUIClass(KEY);
+                uid.put(KEY, val);
             }
         });
+
+        Function<File[], File[]> fun = (Function<File[], File[]>) (File[] files) -> {
+            Favorites favs = Favorites.getDefault();
+            ArrayList<File> list = new ArrayList<>(Arrays.asList(files));
+            for (FileObject root : favs.getFavoriteRoots()) {
+                File file = FileUtil.toFile(root);
+                if (file.isDirectory()) {
+                    list.add(file);
+                }
+            }
+            return list.toArray(File[]::new);
+        };
+        UIManager.put("FileChooser.shortcuts.filesFunction", fun);
```

Comment:
Adopt this as a provided NetBeans key and provide from the Favorites module directly?

Could also consider if Favorites should prepopulate favorites from the shell folders on Windows, and then allow all OS to use Favorites to customise links in file chooser panels for consistency?

Reply:
what is a NetBeans Key?

Reply:
I just meant a UI key that's added and documented in https://github.com/apache/netbeans/blob/master/platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/LFCustoms.java

Reply:
> Could also consider if Favorites should prepopulate favorites from the shell folders on Windows, and then allow all OS to use Favorites to customise links in file chooser panels for consistency?

the way it works right now on windows is that the NB favorites are appended to the windows specific shortcuts:
![image](https://github.com/apache/netbeans/assets/114367/69b0f035-3174-4758-9185-1a7c54234c90)

the win ShellFolder reflection code is essentially not needed in dir file folder if FlatLaf is used since the shortcuts panel is doing this already.